auth_service/src/main/java/vacademy/io/auth_service/feature/auth/filter/InternalAuthFilter.java

```diff
@@ -0,0 +1,60 @@
+package vacademy.io.auth_service.feature.auth.filter;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import vacademy.io.auth_service.feature.auth.service.ClientAuthentication;
+import vacademy.io.auth_service.feature.auth.service.ClientAuthenticationService;
+
+import java.io.IOException;
+import java.util.Base64;
+import java.util.StringTokenizer;
+
+@Component
+public class InternalAuthFilter extends OncePerRequestFilter {
+
+    @Autowired
+    private ClientAuthenticationService clientAuthenticationService;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+            throws ServletException, IOException {
+
+        if (request.getRequestURI().startsWith("/internal/")) {
+            String authorizationHeader = request.getHeader("Authorization");
+
```

Comment:
Let's write the logic of method in the user client service, and keep this filter at each service level that will use the logic